cmake-modules/global_compile_options.cmake

```diff
@@ -13,6 +13,11 @@ if(MSVC)
 
   #https://stackoverflow.com/questions/37527946/warning-unreferenced-inline-function-has-been-removed
   add_compile_options(/W4 ${WARNINGS_AS_ERRORS_FLAG} /wd5031 /wd4668 /wd4820 /wd4255 /wd4710)
+
+  # Since we could be built as either DLL or a static library, when we are built as static library,
+  # linker does generate a bunch of warnings related to the use of dllimport attribute,
+  # basically, suggesting that its use is unneccessary. We know that.
+  add_link_options(/ignore:4217 /ignore:4286)
```

Comment:
Can we put this under a condition, so we only ignore these warnings when building as a static library? Is there a way to detect which flavor of the SDK we are building?

Reply:
nit: In one place where these warnings are ignored, add links to the warning
https://docs.microsoft.com/en-us/cpp/error-messages/tool-errors/linker-tools-warning-lnk4217?view=msvc-160
https://docs.microsoft.com/en-us/cpp/error-messages/tool-errors/linker-tools-warning-lnk4286?view=msvc-160

Or copy the text as a comment:
```cmake
# LNK4217: symbol 'symbol' defined in 'filename_1.obj' is imported by 'filename_2.obj' in function 'function'
# LNK4286: symbol 'symbol' defined in 'filename_1.obj' is imported by 'filename_2.obj'
```

Reply:
Now the warning suppression is not even needed!
We either:
* carry the knowledge the the library was build as DLL via build definitions (fetchcontent case)
* or, for vcpkg, when it is installed, we patch the header. And there is no problem, because VcPkg puts headers from different triplets into different directories.

As a result, it is never the case that you are linking with a static library, but have to deal with a symbol marked as `declspec(dllimport)`. In sttic bilds, they are simply nonexistent, just as before. In fact, they are nonexistent anywhere, but only when Windows DLL gets built.